src/main/java/com/exadel/discountwebapp/location/service/LocationService.java

```diff
@@ -49,14 +54,20 @@ public Location findEntityById(Long id) {
     @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
     public List<CityResponseVO> findAllCitiesByCountryCode(String countryCode) {
         List<Location> locations = locationRepository.findAllByCountry_CountryCode(countryCode);
-        List<CityResponseVO> response = new ArrayList<>();
-        locations.forEach(entity -> response.add(cityMapper.toVO(entity)));
-        return response;
+        List<CityResponseVO> allItems = new ArrayList<>();
+        locations.forEach(entity -> allItems.add(cityMapper.toVO(entity)));
+        return allItems.stream()
+                .filter(el -> el.getCity() != null && !el.getCity().equals("")).distinct()
+                .sorted(Comparator.comparing(CityResponseVO::getCity)).collect(Collectors.toList());
     }
 
     @Transactional(propagation = Propagation.REQUIRED)
     public LocationResponseVO create(LocationRequestVO request) {
-        return locationMapper.toVO(locationRepository.save(locationMapper.toEntity(request)));
+        Location location = locationRepository.findByCountry_CountryCodeAndCityAndAddressLine(
+                request.getCountryCode(),
+                request.getCity(),
+                request.getAddressLine());
+        return location == null ? locationMapper.toVO(locationRepository.save(locationMapper.toEntity(request))) : locationMapper.toVO(location);
```

Comment:
It is more readable to write these methods in separate lines

Reply:
I have used new line, coz if declarate new 2 variables and use them in ternary operator - then we get 500 when entity exits on third time, dunno yet, why. Returned to initial state.